### PR TITLE
Implements proper parsing for the as return type syntax (NOT analyze support)

### PR DESCRIPTION
**Pull request by LemonInTheDark**
I'm not hooking it into the existing return type system because I'm pretty sure it doesn't already support primitives, and I am not prepared to deal with that bullshit.

It will parse without erroring tho, so that's based

### CODE REVIEW

File: crates/dreammaker/src/parser.rs [hunk 1]
```diff
@@ -1152,6 +1144,66 @@ impl<'ctx, 'an, 'inp> Parser<'ctx, 'an, 'inp> {
...
+            return AsType::Anything;
+        };
+        if None == proc_kind {
+            self.error("Cannot specify a return type for a proc override")
+                    .register(self.context);
+        }
+        // While loop over self.next, extracting either the first string with from_name OR
+        // if the first string is a slash, continue to extract, and check for a slash again, and finish with from_vec
+        let next_token: Result<Token, DMError> = self.next("return type");
+        let Ok(ok_token) = next_token else {
```
Comment (SpaceManiac) [inline crates/dreammaker/src/parser.rs:1159]: ditto, possibly more below

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: crates/dreammaker/src/parser.rs [hunk 2]
```diff
@@ -1144,6 +1136,66 @@ impl<'ctx, 'an, 'inp> Parser<'ctx, 'an, 'inp> {
         success((input_type, in_list))
     }
 
+    /// Parse an optional as return type signifier (for procs)
+    fn return_type(&mut self, proc_kind: Option<ProcDeclKind>) -> AsType {
+        if let Ok(None) = self.exact_ident("as") {
```
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply crates/dreammaker/src/parser.rs]: Sure. Refactored the function to use the result pattern, lemme know I got it right

File: crates/dreammaker/src/parser.rs (hunk 1 above)
Comment (LemonInTheDark) [reply crates/dreammaker/src/parser.rs]: handled

File: crates/dreammaker/src/parser.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (SpaceManiac) [reply crates/dreammaker/src/parser.rs]: Looks OK. New conflicts

[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply crates/dreammaker/src/parser.rs]: Copy that

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**